arm/Microsoft.ServiceFabric/clusters/deploy.bicep

```
@@ -0,0 +1,350 @@
+    startPort: contains(nodeType.ephemeralPorts, 'startPort') ? nodeType.ephemeralPorts.startPort : null
+  } : null
+  httpGatewayEndpointPort: contains(nodeType, 'httpGatewayEndpointPort') ? nodeType.httpGatewayEndpointPort : null
+  isPrimary: contains(nodeType, 'isPrimary') ? nodeType.isPrimary : null
+  isStateless: contains(nodeType, 'isStateless') ? nodeType.isStateless : null
+  multipleAvailabilityZones: contains(nodeType, 'multipleAvailabilityZones') ? nodeType.multipleAvailabilityZones : null
+  name: '${(!empty(nodeType.name) ? nodeType.name : 'Node00')}'
+  placementProperties: contains(nodeType, 'placementProperties') ? nodeType.placementProperties : null
+  reverseProxyEndpointPort: contains(nodeType, 'reverseProxyEndpointPort') ? nodeType.reverseProxyEndpointPort : null
+  vmInstanceCount: contains(nodeType, 'vmInstanceCount') ? nodeType.vmInstanceCount : 1
+}]
+
+var notifications_var = [for index in range(0, (!empty(notifications) ? length(notifications) : 0)): {
+  isEnabled: !empty(notifications) ? notifications[index].isEnabled : null
+  notificationCategory: !empty(notifications) ? notifications[index].notificationCategory : null
+  notificationLevel: !empty(notifications) ? notifications[index].notificationLevel : null
+  notificationTargets: !empty(notifications) ? notifications[index].notificationTargets : null
+}]
+
+var reverseProxyCertificate_var = {
+  thumbprint: !empty(reverseProxyCertificate) ? reverseProxyCertificate.thumbprint : null
+  thumbprintSecondary: !empty(reverseProxyCertificate) ? reverseProxyCertificate.thumbprintSecondary : null
+  x509StoreName: !empty(reverseProxyCertificate) ? reverseProxyCertificate.x509StoreName : null
+}
+
+var reverseProxyCertificateCommonNamesList_var = [for index in range(0, (!empty(reverseProxyCertificateCommonNames) ? length(reverseProxyCertificateCommonNames.commonNames) : 0)): {
+  commonNames: reverseProxyCertificateCommonNames.commonNames[index]
+}]
+
+var reverseProxyCertificateCommonNames_var = {
+  commonNames: !empty(reverseProxyCertificateCommonNames) ? reverseProxyCertificateCommonNamesList_var : null
+  x509StoreName: !empty(reverseProxyCertificateCommonNames) ? reverseProxyCertificateCommonNames.x509StoreName : null
+}
+
+var upgradeDescription_var = {
+  deltaHealthPolicy: {
+    applicationDeltaHealthPolicies: !empty(upgradeDescription) ? upgradeDescription.applicationDeltaHealthPolicies : null
+    maxPercentDeltaUnhealthyApplications: !empty(upgradeDescription) ? upgradeDescription.maxPercentDeltaUnhealthyApplications : null
+    maxPercentDeltaUnhealthyNodes: !empty(upgradeDescription) ? upgradeDescription.maxPercentDeltaUnhealthyNodes : null
+    maxPercentUpgradeDomainDeltaUnhealthyNodes: !empty(upgradeDescription) ? upgradeDescription.maxPercentUpgradeDomainDeltaUnhealthyNodes : null
+  }
+  forceRestart: '${!empty(upgradeDescription) ? upgradeDescription.forceRestart : null}'
```

why not 
```suggestion
  forceRestart: !empty(upgradeDescription) ? upgradeDescription.forceRestart : null
```
for this and the others?

---

Updated for forceRestart.
Rest of them require string data type.

---

Agreed, changed everywhere accordingly.